resolver/resolver.go

```diff
@@ -29,9 +29,7 @@ type Resolver struct {
 	rs      *records.RecordGenerator
 	rsLock  sync.RWMutex
 	rng     *rand.Rand
-
-	// pluggable external DNS resolution, mainly for unit testing
-	extResolver exchanger.Exchanger
+	exchanger.Forwarder
```

Comment:
instead of exposing a public field, just declare this as `fwd` and get rid of the confusing utility func (as previously noted)